src/wallet/rpcdump.cpp

```diff
@@ -1282,33 +1287,46 @@ static UniValue ProcessImport(CWallet * const pwallet, const UniValue& data, con
             }
              const CPubKey& pubkey = entry->second;
              CPubKey temp;
-             if (!pwallet->GetPubKey(id, temp) && !pwallet->AddWatchOnly(GetScriptForRawPubKey(pubkey), timestamp)) {
+             if (!pwallet->GetPubKey(id, temp) && !pwallet->AddWatchOnlyWithDB(*batch, GetScriptForRawPubKey(pubkey), timestamp)) {
                 throw JSONRPCError(RPC_WALLET_ERROR, "Error adding address to wallet");
             }
+            if (++cnt % 1000 == 0) {
+                batch.reset(new WalletBatch(pwallet->GetDBHandle()));
+            }
             const auto& key_orig_it = import_data.key_origins.find(id);
             if (key_orig_it != import_data.key_origins.end()) {
-                pwallet->AddKeyOrigin(pubkey, key_orig_it->second);
+                pwallet->AddKeyOriginWithDB(*batch, pubkey, key_orig_it->second);
+                if (++cnt % 1000 == 0) {
+                    batch.reset(new WalletBatch(pwallet->GetDBHandle()));
+                }
             }
             pwallet->mapKeyMetadata[id].nCreateTime = timestamp;
 
             // Add to keypool only works with pubkeys
             if (add_keypool) {
-                pwallet->AddKeypoolPubkey(pubkey, internal);
+                pwallet->AddKeypoolPubkeyWithDB(pubkey, internal, *batch);
+                if (++cnt % 1000 == 0) {
+                    batch.reset(new WalletBatch(pwallet->GetDBHandle()));
+                }
             }
         }
 
         for (const CScript& script : script_pub_keys) {
             if (!have_solving_data || !::IsMine(*pwallet, script)) { // Always call AddWatchOnly for non-solvable watch-only, so that watch timestamp gets updated
-                if (!pwallet->AddWatchOnly(script, timestamp)) {
+                if (!pwallet->AddWatchOnlyWithDB(*batch, script, timestamp)) {
                     throw JSONRPCError(RPC_WALLET_ERROR, "Error adding address to wallet");
                 }
+                if (++cnt % 1000 == 0) {
+                    batch.reset(new WalletBatch(pwallet->GetDBHandle()));
+                }
             }
             CTxDestination dest;
             ExtractDestination(script, dest);
             if (!internal && IsValidDestination(dest)) {
                 pwallet->SetAddressBook(dest, label, "receive");
             }
         }
+        batch.reset();
```

Comment:
Is this necessary? It's about to go out of scope, right?

Reply:
Probably not. I did that just as a belt-and-suspenders thing.